tests/system_tests.py

```diff
@@ -406,42 +408,75 @@ def setUp_file_action(self, expanded_file_name):
         return shutil.copyfile(expanded_file_name, new_name)
 
 
+def test_run(self):
+    """
+    This function reads in the members commands, retval, stdout, stderr and runs
+    the `expand_variables` function on each. The resulting commands are then run
+    using the subprocess module and compared against the expected values that
+    were provided in the static members via `compare_stdout` and
+    `compare_stderr`. Furthermore a threading.Timer is used to abort the
+    execution if a configured timeout is reached.
+
+    It is automatically added as a member function to each system test by the
+    CaseMeta metaclass. This ensures that it is run by each system test
+    **after** setUp() and setUpClass() were run.
+    """
+    for i, command, retval, stdout, stderr in zip(range(len(self.commands)),
+                                                  self.commands,
+                                                  self.retval,
+                                                  self.stdout,
+                                                  self.stderr):
+        command, retval, stdout, stderr = map(
+            self.expand_variables, [command, retval, stdout, stderr]
+        )
+        retval = int(retval)
+        timeout = {"flag": False}
```

Comment:
Does this need to be a dictionary? could it be a simple boolean variable?

Reply:
I tried that when I wrote the first version of the test suite and tried again just to be sure, but it does not work. The issue is, that the timer stops the process after the correct time, but the timeout flag is somehow not set to `True`.

I guess that the problem is that a) the `threading.Timer` runs in another thread, so variable sharing is slightly different and b) a boolean is afaik an immutable type, which behaves differently when passed as an argument to a function. I think this is related to the quirks of Python when passing mutable types like lists, dictionaries, etc. as default parameters to functions, as the list will be shared by all invocations of the function (which can lead to pretty nasty surprises). But I have not investigated this further, as the dict seemed to work.